deployments/google.yml

```diff
@@ -34,37 +40,49 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
-    # Setup gcloud CLI
-    - uses: google-github-actions/setup-gcloud@v0.2.0
+    # Configure Workload Identity Federation and generate an access token.
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0.4.0'
       with:
-        service_account_key: ${{ secrets.GKE_SA_KEY }}
-        project_id: ${{ secrets.GKE_PROJECT }}
+        token_format: 'access_token'
+        workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/my-pool/providers/my-provider'
+        service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
+
+    # Install gcloud CLI
+    - name: 'Install gcloud CLI'
+      uses: 'google-github-actions/setup-gcloud@master'
+
+    # Log in to gcloud CLI
+    - name: Log in to gcloud CLI
+      run: |-
+        gcloud auth login --quiet --brief --cred-file="${{ steps.auth.outputs.credentials_file_path }}"
 
     # Configure Docker to use the gcloud command-line tool as a credential
-    # helper for authentication
-    - run: |-
-        gcloud --quiet auth configure-docker
+    # helper for authentication (https://cloud.google.com/artifact-registry/docs/docker/authentication)
+    - name: Docker configuration
+      run: |-
+        gcloud --quiet auth configure-docker $GAR_LOCATION-docker.pkg.dev
```

Comment:
We can use the access token generated by auth directly rather than via gcloud

something like ` ${{steps.auth.outputs.access_token}} | docker login -u oauth2accesstoken --password-stdin https://$GAR_LOCATION-docker.pkg.dev`

Reply:
Thanks for the example! Do you think it is better to use docker command directly instead of gcloud command?

Reply:
It avoids the need to `setup-gcloud`

Reply:
Sounds reasonable. Added!